### ISSUE TITLE
Why the watcher of  my own  attribute icon print n times when use some icon ?

### ISSUE THREAD

**Issue by bpdarlyn** [xkjyeah/vue-google-maps#242]
``` vue
 <template >
    <google-marker  :position="position" :icon="icon" v-if="showDriver">
    </google-marker>
</template>
<script>
    import {Marker} from 'vue2-google-maps';
    export default {
        props: ["id","position","status","parked","icon"],
        components: { 'google-marker': Marker},
        data(){ return {}},
        mounted(){},
        methods: {
        },
        computed:  {
            showDriver() {
                if (this.status){
                    console.log(this.status);
                    if(this.status.id==1){
                        return true;
                    }else{
                        //todo do something
                    }
                }
                return false;
            }
        },
        watch: {
            'icon.rotation': function(rotation){
                console.log(rotation);
                },
        },
    }
</script>
```

I have a component parent called app-container this has 5 vuecomponent children than are the component shown above and when I change the attribute by console the chrome 
``` 
$vm0.icon.rotation=90
```
The watcher  is executed :
``` vue
watch: {
            'icon.rotation': function(rotation){
                console.log(rotation);
                }
        }
```
but this print 5 times, when should print only once.

My component parent has 5 components of this same, is like if this is executing all watcher the all component of array.  excuse if I you don't understand me.

**Comment by xkjyeah** [comment]
This is a more generic question about Vue so support should come from there.

Also, you shouldn't be modifying the props from the parent content.
